core/domain/user_services.py

```diff
@@ -458,8 +762,19 @@ def update_preferred_site_language_code(user_id, preferred_site_language_code):
 
 def get_human_readable_user_ids(user_ids):
     """Converts the given ids to usernames, or truncated email addresses.
-
     Requires all users to be known.
+
+    Args:
+        user_ids: list(str). The list of user_ids to get UserSettings domain objects for.
+
+    Returns:
+        list(UserSettings). The UserSettings domain objects corresponding to given
+        user ids. If username does not exist, the corresponding entry in the
+        returned list is truncated email address.
+
+    Raises:
+        Exception: If User is not found for any user_id in user_ids list.
```

Comment:
Exception: At least one of the user_ids does not correspond to a valid UserSettingsModel.

(I think that suffices.)